File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/WitchcraftConfig.java
package com.tarbadev.witchcraft;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@EnableConfigurationProperties
@ConfigurationProperties
public class WitchcraftConfig implements WebMvcConfigurer {
    private static final String BACKEND = "/WitchcraftApp";
    private static final String FRONTEND = "/WitchcraftUi";

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        String workingDirectory = System.getProperty("user.dir");

        if (workingDirectory.endsWith(BACKEND)) {
            workingDirectory = workingDirectory.substring(0, workingDirectory.lastIndexOf(BACKEND));
        }

        registry.addResourceHandler("/**").addResourceLocations(
                "classpath:/static/",
                "classpath:/",
                "file:///" + workingDirectory + FRONTEND + "/build/dist/"
        );
    }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/WitchcraftConfig.kt
package com.tarbadev.witchcraft

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.EnableWebMvc
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
@EnableWebMvc
class WitchcraftConfig(
    @Value("\${witchcraft.frontend.url}") private val allowedOrigin: Array<String>
) : WebMvcConfigurer {

  override fun addCorsMappings(registry: CorsRegistry) {
    val corsRegistration = registry.addMapping("/**")
    corsRegistration.allowedOrigins(*allowedOrigin).allowedMethods("GET", "POST", "PUT", "PATCH", "DELETE")
  }
}